src/main/scala/io/iohk/ethereum/utils/Riemann.scala

```diff
@@ -0,0 +1,264 @@
+package io.iohk.ethereum.utils
+
+import io.riemann.riemann.client.RiemannClient
+import io.riemann.riemann.client.Transport
+import io.riemann.riemann.client.EventDSL
+import io.riemann.riemann.client.IRiemannClient
+import io.riemann.riemann.client.IPromise
+import io.riemann.riemann.client.Promise
+import io.riemann.riemann.client.ChainPromise
+import io.riemann.riemann.Proto.Event
+import io.riemann.riemann.Proto.Msg
+import java.net.InetAddress
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.BlockingQueue
+import java.util.LinkedList
+import java.io.IOException
+import scala.collection.JavaConverters._
+import java.util.concurrent.Executors
+import concurrent.{ExecutionContext, Future}
+import com.googlecode.protobuf.format.FormatFactory
+
+trait Riemann extends Logger {
+
+  private val hostName = Config.Network.riemann
+    .map { c =>
+      c.hostName
+    }
+    .getOrElse(InetAddress.getLocalHost().getHostName())
+
+  private val riemannClient = {
+    log.debug("create new RiemannClient")
+    val c = {
+      Config.Network.riemann match {
+        case Some(config) => {
+          log.debug(
+            s"create new riemann batch client connecting to ${config.host}:${config.port}")
+          val client = new RiemannBatchClient(config)
+          try {
+            client.connect()
+            client
+          } catch {
+            case e: IOException =>
+              log.error(e.toString)
+              log.error(
+                "failed to create riemann batch client, falling back to stdout client")
+              new RiemannStdoutClient()
+          }
+        }
+        case None => {
+          log.debug("create new stdout riemann client")
+          new RiemannStdoutClient()
+        }
+      }
+    }
+    c.connect()
```

Comment:
it seems it's called twice for RiemannBatchClient (btw. is this needed for StdoutClient?)

Reply:
its idempotent but fixed anyway since its ugly and bad, thanks